feat(asm): automatic user login events for Django (#5589)
DataDog/dd-trace-py: docs/configuration.rst

   DD_APPSEC_AUTOMATED_USER_EVENTS_TRACKING:
      type: String
      default: "safe"
      description: |
         Sets the mode for the automated user login events tracking feature which sets some traces on each user login event. The
         supported modes are ``safe`` which will only store the user id or primary key, ``extended`` which will also store
         the username, email and full name and ``disabled``. Note that this feature requires ``DD_APPSEC_ENABLED`` to be 
         set to ``true`` to work.  
      version_added:
         v1.15.0:

   DD_USER_MODEL_LOGIN_FIELD:
      type: String
      default: ""
      description: |
         Field to be used to read the user login when using a custom ``User`` model for the automatic login events. Use only if the automatic inference doesn't work.
         Please note that, if set, this field will be user to retrieve the user login even if ``DD_APPSEC_AUTOMATED_USER_EVENTS_TRACKING`` is set to ``safe`` and, 
         in some cases, the selected field could hold potentially private information.
      version_added:
         v1.15.0

   DD_USER_MODEL_EMAIL_FIELD:
      type: String
      default: ""
      description: |

>          Field to be used to read the user email when using a custom ``User`` model for the automatic login events. Use only if the automatic inference doesn't work.

Author: If the env var is set it will be used before automatic inference.

>          Field to be used to read the user email when using a custom ``User`` model for the automatic login events. Use only if the automatic inference doesn't work.

Reviewer: ```suggestion
         Field to be used to read the user email when using a custom ``User`` model for the automatic login events. This field will take precedence over automatic inference.
```

Oh ok got it, what do you think about this phrasing then for both of the envars?

>          Field to be used to read the user email when using a custom ``User`` model for the automatic login events. Use only if the automatic inference doesn't work.

Author: Better, thanks!

      version_added:
         v1.15.0:

   DD_USER_MODEL_NAME_FIELD:
      type: String
      default: ""
      description: |

>          Field to be used to read the user name when using a custom ``User`` model for the automatic login events. Used only if the automatic inference doesn't work.

Reviewer: ```suggestion
         Field to be used to read the user name when using a custom ``User`` model for the automatic login events. This field will take precedence over automatic inference.
```

     version_added:
        v1.13.0:

   DD_APPSEC_AUTOMATED_USER_EVENTS_TRACKING:
      type: String
      default: "safe"
      description: |
         Sets the mode for the automated user login events tracking feature which sets some traces on each user login event. The
         supported modes are ``safe`` which will only stores the user id or primary key, ``extended`` which will also store
         the username, email and full name and ``disabled``. Note that this feature requires ``DD_APPSEC_ENABLED`` to be 
         set to ``true`` to work. For apps using the normal framework authentication models you shouldn't need to configure

>          anything else but in some cases with custom user models you could need to use once of the ``DD_USERMODEL_SOMETHING_FIELD``

Author: The docs were outdated (the RFC changed to only-automatic inference of these fields), removed this, thanks!

         detailed below.
      version_added:
         v1.15.0:

   DD_USERMODEL_LOGIN_FIELD:
      type: String
      default: ""
      description: |
         Sets the field of the authentication user model to use for retrieving the user id or username. This will be used in the
         automatic user login events feature (see ``DD_APPSEC_AUTOMATED_USER_EVENTS_TRACKING`` above). Use it only if the automatic
         inference doesn't work. Please note that if you set this variable, the tracer will try to use whatever field you set

>          even if ``safe`` mode is enabled!

Author: Ditto. Outdated.
